### PR TITLE
feat: add oci image layout mutation 

### LINKED ISSUE #1433: crane: mutate OCI image layout with annotations, layers etc.
### Describe the bug

This is not a bug, and this is a feature.

### Additional context

Recently, @sudo-bmitch talked about OCI image layout[^1]. He showed a demo of making file operations on the OCI image layout directory for adding new blobs, annotations, etc. crane already has the support of working with OCI image layout through a package `pkg/v1/layout` available in go-containerregistry but editing the OCI image layout for such operations adding new blobs, annotations, etc. is not possible, so, we talked with @imjasonh about this, and `mutate` cmd is the best place to implement this. Based on that, I've tested a sample code to make changes to OCI image layout, and it worked:

```golang
img2 := mutate.Annotations(img, annotations).(v1.Image)
err = p.ReplaceImage(img2, match.Digests(digest))
if err != nil {
	return fmt.Errorf("replacing image: %w", err)
}
```

[^1]: https://www.youtube.com/watch?v=TJVoFKTm9I0

PTAL @Dentrax

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>
Co-authored-by: Furkan Türkal <furkan.turkal@trendyol.com>

Fixes #1433 

kindly ping @imjasonh for review

### CODE REVIEW

File: cmd/crane/recipes.md [hunk 1]
```diff
@@ -79,3 +79,11 @@ crane manifest gcr.io/buildpacks/builder:v1 | jq '.config.size + ([.layers[].siz
 ```
 
 For image indexes, you can pass the `--platform` flag to `crane` to get a platform-specific image.
+
+### Add Annotations
```
Comment (imjasonh) [inline cmd/crane/recipes.md:83]: This isn't exactly a recipe, this is just how to use the tool. It's not very well defined, but I normally think of a recipe as how to use 2+ things together, like how to pipe the output of crane extract to tar, for example.

Comment (developer-guy) [reply cmd/crane/recipes.md]: wrong commit :/ I've removed it.

**Comment by developer-guy** [comment]
Okay, now it's working fine, thanks to @Dentrax.

You can test this with the following commands:

```shell
$ crane pull --format oci devopps/hello-world:buildinfo hello-world-oci-layout
$ tar czf layer.tar.gz README.md
$ LAYER_DIGEST=$(sha256sum layer.tar.gz | cut -d' ' -f1) 
$ ./crane mutate --append layer.tar.gz --oci-image-layout hello-world-oci-layout
$ tar tvf hello-world-oci-layout/blobs/sha256/$LAYER_DIGEST
-rw-r--r--  0 batuhan.apaydin staff    2968 Aug 23 22:59 README.md
```

**Comment by developer-guy** [comment]
kindly ping @imjasonh @jonjohnsonjr

**Comment by developer-guy** [comment]
kindly ping folx @imjasonh @jonjohnsonjr ☝️

**Comment by developer-guy** [comment]
kindly ping folx @imjasonh @jonjohnsonjr 🫣

**Pull request closed without merging**